### ISSUE TITLE
Searching

### ISSUE THREAD

**Issue by andrewheiss** [jakob/TableTool#7]
It would be cool if TableTool had some sort of basic search capability, even if it just searched the whole document (and didn't have options to limit searches by columns or whatever).

**Comment by pgensler** [comment]
Yes, this would be extremely helpful, to search a dataset by row number or column to find errors.
This is a great use case file below: "http://daniels-pull.universityofdenv.netdna-cdn.com/assets/GeneralOccurrencesAll.csv"

**Comment by anadin** [comment]
+1 for search would be super handy

**Comment by emartin-pentaho** [comment]
+1. The app has a nice UI and handy features, but being able to search for strings would push it over the edge for me.

**Comment by StephanJanosch** [comment]
+1

**Comment by oberlage** [comment]
+1

**Comment by eauw** [comment]
+1

**Comment by fulldecent** [comment]
Will somebody please propose a sample user interface for search. This will help continue the discussion and get closer to a solution.

**Comment by andrewheiss** [comment]
For column-based searching and/or filtering, it could be helpful to do something similar to RStudio, which has search boxes above each column and then only displays rows that match the conditions:

<img width="297" alt="screen shot 2018-07-05 at 14 08 30" src="https://user-images.githubusercontent.com/73663/42345605-1750c286-805d-11e8-86fe-147f4b82c8c3.png">

<img width="295" alt="screen shot 2018-07-05 at 14 08 38" src="https://user-images.githubusercontent.com/73663/42345614-1c50646c-805d-11e8-989b-593c00ba3fbe.png">

For document-wide searching, following Excel could be helpful, with a search field in top area of the GUI:

<img width="923" alt="screen shot 2018-07-05 at 14 11 54" src="https://user-images.githubusercontent.com/73663/42345708-725f30cc-805d-11e8-865f-9ecc613675f4.png">

**Comment by inovek** [comment]
+1

**Comment by nikitavoloboev** [comment]
Anyone working on this? I can try add it because without search this tool is not that useful.

**Comment by fulldecent** [comment]
@nikitavoloboev PR please
